src/layers/stateless.jl

```diff
@@ -2,9 +2,88 @@ using CuArrays
 using NNlib: logsoftmax, logσ
 
 # Cost functions
+"""
+    mae(ŷ, y)
+L1 loss function. Computes the mean of absolute error between prediction and true values
+"""
+mae(ŷ, y) = sum(abs.(ŷ .- y)) * 1 // length(y)
+
 
+"""
+    mse(ŷ, y)
+L2 loss function. Computes the mean of the squared errors between prediction and true values
+"""
 mse(ŷ, y) = sum((ŷ .- y).^2) * 1 // length(y)
 
+
+"""
+    msle(ŷ, y;ϵ1=eps.(Float64.(ŷ)),ϵ2=eps.(Float64.(y)))
+
+Mean Squared Logarithmic Error,an L2 loss function. Returns the mean of the squared logarithmic errors of prediction ŷ, and true values y. The ϵ1 and ϵ2 terms provide numerical stability.
+(Computes mean of squared(log(predicted values)-log(true value)). This error penalizes an under-predicted estimate greater than an over-predicted estimate.
+
+  ```julia
+  julia> y=[14726,327378,74734]
+  3-element Array{Int64,1}:
+    14726
+  327378
+    74734
+
+  julia> ŷ = [12466.1,16353.95,16367.98]
+  3-element Array{Float64,1}:
+  12466.1 
+  16353.95
+  16367.98
+
+  julia> msle(ŷ,y)
+  3.771271382334686
+  ```
+"""
+msle(ŷ, y;ϵ1=eps.(ŷ),ϵ2=eps.(eltype(ŷ).(y))) = sum((log.(ŷ+ϵ1).-log.(y+ϵ2)).^2) * 1 // length(y)
+
+
+
+"""
+    huber_loss(ŷ, y,delta=1.0)
+
+Computes the mean of the Huber loss between prediction ŷ and true values y. By default, delta is set to 1.0.
+[Huber Loss](https://en.wikipedia.org/wiki/Huber_loss).
+  
+  ```julia
+  julia> y = [1.2636,1.25,1.73]
+  3-element Array{Float64,1}:
+  1.2636
+  1.25  
+  1.73  
+
+  julia> y_= [-1.376,0,3.37]
+  3-element Array{Float64,1}:
+  -1.376
+   0.0  
+   3.37 
+
+  julia> huber_loss(y,y_)
+  0.7131999999999998
+  ```
+
+"""
+function huber_loss(ŷ, y,delta=1.0)
+  abs_error = abs.(ŷ.-y)
+  dtype= eltype(ŷ)
+  delta = dtype(delta)
+  hub_loss = dtype(0)
+  for i in 1:length(y)
+    if (abs_error[i]<=delta)
+      hub_loss+=abs_error[i]^2*dtype(0.5)
+    else
+      hub_loss+=delta*(abs_error[i]- dtype(0.5*delta))
+    end
+  
+  return hub_loss*1//length(y)
+  end
```

Comment:
The return statement should not be inside the inner loop?
And I think that Julia ppl prefer to omit the 'return' keyword
```suggestion
  end
  hub_loss*1//length(y)
```

Reply:
Oh yes!, that should have been outside the for loop. Thanks for pointing.